src/charm.py

```diff
@@ -788,6 +799,20 @@ def _get_cpu_extensions() -> list[str]:
                 del cpu_flags[0]
         return cpu_flags
 
+    def _cpu_is_compatible_for_hugepages(self) -> bool:
+        return all(
+            required_extension in self._get_cpu_extensions()
+            for required_extension in REQUIRED_CPU_EXTENSIONS_HUGEPAGES
+        )
+
+    @staticmethod
+    def _hugepages_are_available() -> bool:
+        client = Client()
+        nodes = client.list(Node)
+        if not nodes:
+            return False
+        return all([node.status.capacity.get("hugepages-1Gi", "0") >= "2Gi" for node in nodes])
```

Comment:
I'd suggest using `allocatable` instead of `capacity`. Some part of `capacity` will be reserved by the system, kubelet, etc.

Reply:
I think checking node capacity is fine here. Because If the node does not have 2Gi capacity, the charm will go to Blocked state.  However, the mode may have capacity but not enough resource. In this case, the bessd container will not start and the pod will be in waiting status.

Reply:
Yes, in this case it's not critical, but in general, I'd rather use `allocatable`.